marimo/_server/model.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Marimo. All rights reserved.
+import abc
+from enum import Enum
+from multiprocessing.connection import Connection
+from typing import Any, Callable
+
+
+class ConnectionState(Enum):
+    """Connection state for a session"""
+
+    OPEN = 0
+    CLOSED = 1
+
+
+class SessionMode(Enum):
+    """Session mode for a session"""
+
+    # read-write
+    EDIT = 0
+    # read-only
+    RUN = 1
+
+
+class SessionHandler(metaclass=abc.ABCMeta):
```

Comment:
this abstraction isn't perfect, but does remove the coupling to torando. the arguments passed should be sufficient for other connection types